src/master-playlist-controller.js

```diff
@@ -548,7 +549,34 @@ export class MasterPlaylistController extends videojs.EventTarget {
     this.mainSegmentLoader_.on('bandwidthupdate', () => {
       // figure out what stream the next segment should be downloaded from
       // with the updated bandwidth information
-      this.masterPlaylistLoader_.media(this.selectPlaylist());
+      const nextPlaylist = this.selectPlaylist();
+      const currentPlaylist = this.masterPlaylistLoader_.media();
+      const buffered = this.tech_.buffered();
+      const forwardBuffer = buffered.length ?
+        buffered.end(buffered.length - 1) - this.tech_.currentTime() : 0;
+
+      const currentTime = this.tech_.currentTime();
+      const initial = Config.BUFFER_LOW_WATER_LINE;
```

Comment:
We should make getter functions on MPC for low water line and goal buffer length, and pass the GBL one down to the segment loader to use